ndb_api/db_helpers.go

```diff
@@ -142,4 +109,37 @@ func GenerateDeprovisionDatabaseRequest() (req *DatabaseDeprovisionRequest) {
 		DeleteLogicalCluster: true,
 	}
 	return
+
+}
+
+func validateReqData(ctx context.Context, databaseInstanceType string, reqData map[string]interface{}) (err error) {
+	log := ctrllog.FromContext(ctx)
+	dbPassword, ok := reqData[common.NDB_PARAM_PASSWORD].(string)
+	// Type Assertion for dbPassword
+	if !ok || dbPassword == "" {
+		err = errors.New("invalid database password")
```

Comment:
Curious when the "type assertion" may fail for password? If we do not get into that case, the error message can simply be updated to "empty database password"

Reply:
Discussed this offline. Since "map[string]interface{}" can accept any datatype (string, int, map, object etc) as "value", we need to assert.